Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean

```diff
@@ -47,10 +55,98 @@ rectangle are contained in `s` by convexity. The general case follows by lineari
 -/
 
 
-open Asymptotics Set
+open Asymptotics Set Filter
 
 open scoped Topology
 
+section General
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E F : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NormedAddCommGroup F]
+  [NormedSpace 𝕜 F] {s t : Set E} {f : E → F} {x : E}
+
+variable (𝕜) in
+/-- Definition recording that a function has a symmetric second derivative within a set at
+a point. This is automatic in most cases of interest (open sets over real or complex vector fields,
+or general case for analytic functions), but we can express theorems of calculus using this
+as a general assumption, and then specialize to these situations. -/
+def IsSymmSndFDerivWithinAt (f : E → F) (s : Set E) (x : E) : Prop :=
+  ∀ v w, fderivWithin 𝕜 (fderivWithin 𝕜 f s) s x v w = fderivWithin 𝕜 (fderivWithin 𝕜 f s) s x w v
+
+variable (𝕜) in
+/-- Definition recording that a function has a symmetric second derivative within a set at
+a point. This is automatic in most cases of interest (open sets over real or complex vector fields,
+or general case for analytic functions), but we can express theorems of calculus using this
+as a general assumption, and then specialize to these situations. -/
+def IsSymmSndFDerivAt (f : E → F) (x : E) : Prop :=
```

Comment:
Could you please add the lemma that projects from this predicate to `IsSymmSndFDerivWithinAt`?

Reply:
Done.